### DISCUSSION TITLE
What's the difference between using preset-typescript vs directly using plugin-transform-typescript?

### DISCUSSION THREAD

**Discussion by JoshRobertson** [babel/babel#12952] [Q&A] [answered]
The documentation makes it seem like there's nothing in the preset other than the one plugin. Are there undocumented reasons why the preset is preferred (or at least different)?

https://babeljs.io/docs/en/babel-preset-typescript
https://babeljs.io/docs/en/babel-plugin-transform-typescript

**Comment by nicolo-ribaudo** [comment, marked as answer]
The main difference is that the preset only enables the TS plugin on files ending with `.ts`, while the TS plugins doesn't have this mechanism (you have to "manually" configure it to only run on `.ts` files _if you need it_).

**Reply by JoshRobertson** [reply]
cool, good to know!
